airframe-http-recorder/src/main/scala/wvlet/airframe/http/recorder/HttpRecorder.scala

```diff
@@ -106,24 +114,27 @@ object HttpRecorder extends LogSupport {
     */
   def createReplayOnlyServer(recorderConfig: HttpRecorderConfig): FinagleServer = {
     val recorder = new HttpRecordStore(recorderConfig)
+    // Return the server instance as FinagleServer to avoid further recording
     new HttpRecorderServer(recorder, HttpRecorderServer.newReplayService(recorder))
   }
 
   /**
     * Creates an HTTP server that returns programmed HTTP responses.
     * If no matching record is found, use the given fallback handler.
     */
-  def createProgrammableServer(programmer: HttpRecordStore => Unit): FinagleServer = {
-    val recorderConfig = HttpRecorderConfig("localhost")
-    val recorder = new HttpRecordStore(recorderConfig, true) {
-      override def requestHash(request: Request): Int = {
-        computeRequestHash(request, recorderConfig)
-      }
-    }
-    programmer(recorder)
+  def createProgrammableServer(recorderConfig: HttpRecorderConfig): HttpRecorderServer = {
```

Comment:
Changed the interface to accept a custom config. To program responses, use HttpRecorderServer.record or recordIfNotExists